### PR TITLE
Improve `write_timeout` Handling for Media Methods

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Second part of #3864

Unfortunately I realised that the approach outlined in #3864 is not as backward compatible as I initially thought, but I think I found a rather clean deprecation layer

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- ~Added myself alphabetically to ``AUTHORS.rst`` (optional)~
-  ~Added new classes & modules to the docs and all suitable ``__all__`` s~
-  [x] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

### CODE REVIEW

**Review by Poolitzer — APPROVED** [review]
LGTM

File: docs/auxil/sphinx_hooks.py [hunk 1]
```diff
@@ -107,19 +107,31 @@ def autodoc_process_docstring(
...
-                keyword_args[i - insert_index].format(
+            to_insert = keyword_args[i - insert_index]
+
+            if (
+                "post.write_timeout`. Defaults to" in to_insert
+                and method_name in media_write_timeout_deprecation_methods
+            ):
+                effective_insert = media_write_timeout_deprecation
+            else:
+                effective_insert = (to_insert,)
```
Comment (harshil21) [inline docs/auxil/sphinx_hooks.py:122]:
```suggestion
                effective_insert: list[str] = media_write_timeout_deprecation
            else:
                effective_insert: list[str] = list(to_insert)
```
unified types would be nice and could avoid potential bugs in the future

File: docs/auxil/kwargs_insertion.py [hunk 2]
```diff
@@ -45,7 +45,33 @@
...
+    "create_new_sticker_set",
+]
+media_write_timeout_deprecation = [
+    "    write_timeout (:obj:`float` | :obj:`None`, optional): Value to pass to "
+    "        :paramref:`telegram.request.BaseRequest.post.write_timeout`. Defaults to ``20``."
+    "",
+    "",
+    "       .. deprecated:: NEXT.VERSION",
+    "           In future versions, the default value will be changed to "
+    "           :attr:`~telegram.request.BaseRequest.DEFAULT_NONE`.",
```
Comment (harshil21) [inline docs/auxil/kwargs_insertion.py:71]: technically the parameter was already changed to be `DEFAULT_NONE`  so we can't say "in future versions"

**Review by harshil21 — COMMENTED** [review]
logic looks good, just few minor comments

File: docs/auxil/kwargs_insertion.py (hunk 2 above)
Comment (Bibo-Joshi) [reply docs/auxil/kwargs_insertion.py:71]: mh, I don't really mean the value behing the "=" in the signature but rather the value that will be passed to `BaseRequest.post`. And that's still 20s after this PR. I tried to specify a bit

**Review by harshil21 — APPROVED** [review]

**Pull request merged**